consensus/replay.go

```diff
@@ -456,11 +465,26 @@ func (h *Handshaker) replayBlock(state sm.State, height int64, proxyApp proxy.Ap
 	return state, nil
 }
 
-func checkAppHash(state sm.State, appHash []byte) error {
-	if !bytes.Equal(state.AppHash, appHash) {
-		panic(fmt.Errorf("Tendermint state.AppHash does not match AppHash after replay. Got %X, expected %X", appHash, state.AppHash).Error())
+func checkAppHashAgainstBlock(appHash []byte, block *types.Block) {
```

Comment:
The name is slightly misleading. This isn't a simple check anymore as we only panic. Don't know what is the best name here to signal this but I think `assertXXX`works.  E.g.
`assertAppMatchesBlockHash`
`assertAppMatchesStateHash`

You could even combine this in a single method:
`assertAppHashEq(appHash []byte, otherHash []byte, panicMsg string)`

Reply:
Updated. Thanks!